src/holofuel/components/layout/PrimaryLayout/PrimaryLayout.js

```diff
@@ -19,23 +18,28 @@ export function PrimaryLayout ({
   headerProps = {}
 }) {
   const { data: { holofuelActionableTransactions: transactions = [] } = {} } = useQuery(HolofuelActionableTransactionsQuery)
-  const { data: { holofuelUser = {} } = {} } = useQuery(HolofuelUserQuery)
+  const { loading: holofuelUserLoading, data: { holofuelUser = {} } = {} } = useQuery(HolofuelUserQuery)
   const { data: { holofuelLedger: { balance: holofuelBalance } = { balance: 0 } } = {} } = useQuery(HolofuelLedgerQuery)
 
   const inboxCount = transactions.length
-  const agentId = presentAgentId(holofuelUser.id)
+
+  let agent = {}
+  let agentLoading
+  if (holofuelUserLoading) agentLoading = true
+  else agent = { ...holofuelUser }
```

Comment:
this doesn't need to be conditional. You can always assign `agent` to `holofuelUser` here, because `holofuelUser` will be empty by default.

In fact, I'm pretty sure lines 26 - 29 can be removed if you change line 37 to
`<Header {...headerProps} agent={holofuelUser} agentLoading={holofuelUserLoading} hamburgerClick={hamburgerClick} />`

and line 42 to
`agentLoading={holofuelUserLoading}`